File: 프로그래머스/2/42888. 오픈채팅방/오픈채팅방.java
import java.util.*;
class Solution {
    static Map<String, String> id_map;
	static Queue<String> chat_queue;
    public String[] solution(String[] record) {
    	id_map = new HashMap<>();
    	chat_queue = new LinkedList<>();
        String[] answer = {};
        List<String> answer_list = new ArrayList<>();
        for(int i = 0; i < record.length; i++) {
        	String[] str_arr = record[i].split(" ");
        	//Leave는 최대 Index가 1이다.
        	if(str_arr[0].equals("Leave")) {
        		chat_queue.add(str_arr[0]+" "+str_arr[1]);
        	}else {
            	id_map.put(str_arr[1], str_arr[2]);
            	// Change냐 Enter냐만 구분하여 넣어놓기
            	if(str_arr[0].equals("Change"))
            		continue;
            	else
            		chat_queue.add(str_arr[0]+" "+str_arr[1]);
        	}
        }
        while(!chat_queue.isEmpty()) {
        	String answer_str = chat_queue.poll();
        	String[] str_arr = answer_str.split(" ");
        	if(str_arr[0].equals("Enter")) {
        		answer_list.add(id_map.get(str_arr[1])+"님이 들어왔습니다.");
        	}else {
        		answer_list.add(id_map.get(str_arr[1])+"님이 나갔습니다.");
        	}
        }
        answer = answer_list.toArray(answer);
        return answer;
    }
}

File: 프로그래머스/2/42888. 오픈채팅방/오픈채팅방.js
function solution(record) {
  const answer = [];
  const userInfo = {};
  //userInfo를 {} 객체로 생성하여 이름과 값 쌍을로 구성 -> 이게 낫네... 괜히 객체배열로 하느니...

  for (let i = 0; i < record.length; i++) {
    const list = record[i].split(" ");
    if (list[0] === "Enter") {
      userInfo[list[1]] = list[2];
      answer.push(`${list[1]}님이 들어왔습니다.`);
    } else if (list[0] === "Leave") {
      answer.push(`${list[1]}님이 나갔습니다.`);
    } else if (list[0] === "Change") {
      userInfo[list[1]] = list[2];
    }
  }

  // replace로 이름으로 변경하기
  for (let i = 0; i < answer.length; i++) {
    const uid = answer[i].split("님이")[0];
    answer[i] = answer[i].replace(uid, userInfo[uid]);
  }

  return answer;
}